File: sdk/kotlin/src/wasmWasiMain/kotlin/functionfly/Function.kt
package functionfly

/**
 * Interface that all FunctionFly Kotlin functions must implement.
 */
fun interface Function {
    /**
     * Process the input and return the output.
     * @param input The function input as a JSON string
     * @return The function output as a JSON string
     */
    fun handle(input: String): String
}

File: sdk/swift/Sources/FunctionFly/Function.swift
/// FunctionFly Swift SDK — Function protocol.
///
/// Implement this protocol to create a FunctionFly function:
///
///     struct MyFunction: FunctionFlyFunction {
///         func handle(input: String, context: FunctionFlyContext) throws -> String {
///             return #"{"message": "Hello from Swift!"}"#
///         }
///     }
///
///     FunctionFly.run(MyFunction())
import Foundation

/// Protocol that all FunctionFly functions must implement.
public protocol FunctionFlyFunction {
    /// Process the input and return the output.
    func handle(input: String, context: FunctionFlyContext) throws -> String
}
